src/main/java/org/alfresco/rest/api/impl/QuickShareLinksImpl.java

```diff
@@ -581,28 +589,48 @@ private QuickShareLink getQuickShareInfo(NodeRef nodeRef, Map<String, Object> ma
             // note: if noAuth mode then do not return allowable operations (eg. but can be optionally returned when finding shared links)
             if (!noAuth)
             {
+                // Cached document
+                Node doc = null;
+
                 if (includeParam.contains(PARAM_INCLUDE_ALLOWABLEOPERATIONS))
                 {
                     if (quickShareService.canDeleteSharedLink(nodeRef, sharedByUserId))
                     {
                         // the allowable operations for the shared link
-                    qs.setAllowableOperations(Collections.singletonList(Nodes.OP_DELETE));
-                }
-
-                    Node doc = nodes.getFolderOrDocument(nodeRef, null, null, includeParam, null);
+                        qs.setAllowableOperations(Collections.singletonList(Nodes.OP_DELETE));
+                    }
+                    doc = getNode(doc, nodeRef, includeParam, mapUserInfo);
                     List<String> allowableOps = doc.getAllowableOperations();
                     // the allowable operations for the actual file being shared
                     qs.setAllowableOperationsOnTarget(allowableOps);
                 }
 
-                
                 // in noAuth mode we don't return the path info
                 if (includeParam.contains(PARAM_INCLUDE_PATH))
                 {
                     qs.setPath(nodes.lookupPathInfo(nodeRef, null));
                 }
-            }
 
+                if (includeParam.contains(PARAM_INCLUDE_PROPERTIES))
+                {
+                    doc = getNode(doc, nodeRef, includeParam, mapUserInfo);
```

Comment:
We should avoid invoking this multiple times and check if doc is already != null